doc/rst/language/spec/domains.rst

```diff
@@ -98,19 +98,24 @@ may make concurrent queries and iterations on a domain as long as
 another task is not simultaneously modifying the domain's index
 set.
 
-By default, associative domains permit multiple tasks
-to modify their index sets concurrently.  This adds some amount of
-overhead to these operations.  If the user knows that all such
-modifications will be done serially or in a parallel-safe context,
-the overheads can be avoided by setting ``parSafe`` to ``false`` in
-the domain's type declaration.  For example, the following
-declaration creates an associative domain of strings where the
-implementation will do nothing to ensure that simultaneous
-modifications to the domain are parallel-safe:
+By default, associative domains do not permit multiple tasks
+to modify their index sets concurrently without race conditions.
+This setting is controlled by the ``parSafe`` parameter of the domain type,
+which defaults to ``false``. This is because ``parSafe`` uses locking on
+the underlying data structure each time the domain is modified. This
+overhead is unnecessary, for example, when the domain is operated upon by a
+single task.
+
+Setting ``parSafe`` to ``true`` allows multiple tasks to modify the index set
+of an associative domain concurrently. This might be useful when the domain is
+operated upon by multiple tasks. However, it is the user's responsibility to
+ensure that the domain is not accessed while its index set is being modified.
+The following example demonstrates how to create a parallel-safe associative
+domain of strings:
```

Comment:
Reading the above two paragraphs: they start with a repetition of what was said just prior. I would like to avoid this repetition. Instead, I would like to start right away with what is different about the associative domains. Here is my draft:

An associative domain may permit multiple tasks to modify its index set concurrently in a parallel-safe manner if its type is declared with \`\`parSafe=true\`\`.  For example:

... the string domain example ...

Note that declaring a domain with \`\`parSafe=true\`\` adds some amount of overhead to many domain operations. This is because such a domain uses locking on the underlying data structure each time the domain is modified. This overhead is unnecessary, for example, when the domain is operated upon only by a single task, in which case it can be avoided by declaring the domain's type with \`\`parSafe=false\`\` or without an explicit \`\`parSafe\`\` setting.

Note that the \`\`parSafe=true\`\` mode is currently unstable. Its availability and behavior may change in the future.

As with any other domain type, ...... [continue as before]